cms/static/js/views/metadata.js

```diff
@@ -220,6 +225,44 @@ function(BaseView, _, MetadataModel, AbstractEditor, FileUpload, UploadDialog, V
 
     });
 
+    Metadata.NumberWithWarning = Metadata.Number.extend({
+        initialize: function() {
+            Metadata.Number.prototype.initialize.apply(this);
+            var options = this.model.get("options");
+            this.msg = options["warning"];
+            // show initial warning if necessary
+            this.checkWarning();
+        },
+
+        changed: function() {
+            Metadata.Number.prototype.changed.apply(this);
+            this.checkWarning();
+        },
+
+        checkWarning: function() {
+            var value = this.getValueFromEditor();
+            if (parseInt(value)) {
```

Comment:
thanks for catching the missing radix . I'm checking here whether the value stored in "value" is interpretable as an int. `parseInt` should return NaN if not, a falsy value. Should I explicitly check (use isNaN) here?